src/plugins/intel_gpu/tests/test_cases/topology_test.cpp

```diff
@@ -421,46 +437,41 @@ class topology_test : public ::testing::TestWithParam<topology_params>
 
         return ss.str();
     }
+
+    static std::shared_ptr<cldnn::engine> get_engine(){
+        if(!engine){
+            //getting pointer of return value of tests::get_test_engine() is safe because it's memory is allocated in static memory.
+            engine=std::shared_ptr<cldnn::engine>(&tests::get_test_engine());
+        }
+        return engine;
+    }
 protected:
     cldnn::layout output_layout;
     std::vector<unsigned> generator;
 
-    static cldnn::engine& engine;
+    static std::shared_ptr<cldnn::engine> engine;
     static std::vector<cldnn::layout> all_output_layouts;//just for tear-down
 };
 
-cldnn::engine& topology_test::engine = tests::get_test_engine();
+std::shared_ptr<cldnn::engine> topology_test::engine;
 std::vector<cldnn::layout> topology_test::all_output_layouts = {};
-
-std::vector<std::shared_ptr<topology_test::topology_generator::topology_layer_type>> topology_test::topology_generator::layer_types = {
-    std::shared_ptr<topology_test::topology_generator::topology_layer_type>(new topology_test::topology_generator::normalization_layer_type()),
-    std::shared_ptr<topology_test::topology_generator::topology_layer_type>(new topology_test::topology_generator::pooling_layer_type()),
-    std::shared_ptr<topology_test::topology_generator::topology_layer_type>(new topology_test::topology_generator::convolution_layer_type()),
-    std::shared_ptr<topology_test::topology_generator::topology_layer_type>(new topology_test::topology_generator::fully_connected_layer_type()),
-    std::shared_ptr<topology_test::topology_generator::topology_layer_type>(new topology_test::topology_generator::reorder_layer_type()),
-    std::shared_ptr<topology_test::topology_generator::topology_layer_type>(new topology_test::topology_generator::activation_layer_type()),
-    std::shared_ptr<topology_test::topology_generator::topology_layer_type>(new topology_test::topology_generator::depth_concatenate_layer_type()),
-    std::shared_ptr<topology_test::topology_generator::topology_layer_type>(new topology_test::topology_generator::eltwise_layer_type()),
-    std::shared_ptr<topology_test::topology_generator::topology_layer_type>(new topology_test::topology_generator::softmax_layer_type()),
-    // Only add new types at the end
-};
 const cldnn::primitive_id topology_test::topology_generator::output_layer_id("tg_output_layer");
 
 TEST_P(topology_test, TOPOLOGY)
 {
-     try
-     {
-         run_single_test();
-         if (::testing::Test::HasFailure())
-         {
-             PrintTupleTo(GetParam(), &std::cout);
-         }
-     }
-     catch (...)
-     {
-         PrintTupleTo(GetParam(), &std::cout);
-         throw;
-     }
+    try
+    {
+        run_single_test();
+        if (::testing::Test::HasFailure())
+        {
+            PrintTupleTo(GetParam(), &std::cout);
+        }
+    }
+    catch (...)
+    {
+        PrintTupleTo(GetParam(), &std::cout);
+        throw;
+    }
```

Comment:
@vladimir-paramuzov  I checked this case itself. Even without this PR, no test case is instantiated from this. Can we just remove this case?

```
[ov1] mingyuki@dg2alderlake:/mnt/local/mingyuki/ov1/bin/intel64/Release$ ./clDNN_unit_tests64 --gtest_list_tests | grep -i topology
  DISABLED_shared_mem_pool_same_topology_twice
  DISABLED_shared_mem_pool_same_topology_twice_weights
```

Reply:
Deleted topology_test.cpp